habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1218,6 +1224,12 @@ def step(
 
         return self._get_agent_observation(agent_state_result)
 
+    def _check_stop(self, linear_velocity, angular_velocity) -> bool:
+        """Use a heuristic to determine if an action qualifies as a stop action"""
+        lin_check = abs(linear_velocity) < self._config.min_abs_lin_speed
```

Comment:
Avoid using hydra configs within a loop. This will cause considerable slow down as noted by Vince in a prior PR.

Reply:
Wow that's new to me. Interesting design choice from Hydra to have configs resolved dynamically.
Will fix.

Reply:
@ykarmesh Can you provide a link to the PR you mentioned? Would love to read more on this matter, because it would affect certain designs in my other projects too.

Reply:
https://github.com/facebookresearch/habitat-lab/pull/1120